OpenRA.Mods.Common/Traits/CreatesShroud.cs

```diff
@@ -18,10 +18,24 @@ public class CreatesShroudInfo : AffectsShroudInfo
 
 	public class CreatesShroud : AffectsShroud
 	{
+		readonly Actor self;
+
 		public CreatesShroud(Actor self, CreatesShroudInfo info)
-			: base(self, info) { }
-		protected override void AddCellsToPlayerShroud(Actor self, Player p, PPos[] uv) { p.Shroud.AddProjectedShroudGeneration(self, uv); }
-		protected override void RemoveCellsFromPlayerShroud(Actor self, Player p) { p.Shroud.RemoveShroudGeneration(self); }
+			: base(self, info) { this.self = self; }
+
+		protected override void AddCellsToPlayerShroud(object o, Player p, PPos[] uv)
```

Comment:
You can avoid most of the changes in these two files by keeping this as `(Actor self, Player p, PPos[] uv)` and then passing doing `AddProjectedShroudGeneration(this, uv)`